src/ICBOT/utils/meteo.py

```diff
@@ -39,25 +38,47 @@ def __str__(self) -> str:
         return Messages.METEO.format_map(vars(self))
 
 
-class _WeatherCacher:
+class KeyValueCache:
+    """
+    A simple cache that stores values for given keys in memory.
+    """
+
     def __init__(self) -> None:
-        self._timestamp = datetime.datetime.min
+        self._values = {}  # key → value
+        self._expirations = {}  # key → DateTime of the expiration
 
-    def cache(self, weathers: List[WeatherEntry]):
-        self._cached = weathers
-        self._timestamp = datetime.datetime.now()
-        return weathers
+    def cache(self, key, value, seconds):
+        """
+        Stores a new value in the cache.
 
-    def value(self):
-        return self._cached
+        :param key: the key
+        :param value: the value
+        :param seconds: how many seconds the value lasts before it needs to be replaced
+        :returns: the value
+        """
 
-    def needs_refresh(self):
-        return (datetime.datetime.now() - self._timestamp) > datetime.timedelta(
-            hours=Constants.REFRESH_HOURS_WEATHER
-        )
+        self._values[key] = value
+        self._expirations[key] = datetime.datetime.now() + datetime.timedelta(seconds=seconds)
+        return value
+
+    def get(self, key):
+        """
+        Returns the latest stored value for a key, or None if no value was stored.
+
+        :param key: the key
+        """
 
+        return self._values[key] if key in self._values else None
```

Comment:
You can use built-in dict's [get](https://docs.python.org/3/library/stdtypes.html#dict.get) instead to do specifically this.